src/PaymentGateways/Stripe/ApplicationFee.php

```diff
@@ -28,14 +29,19 @@ class ApplicationFee {
 	 *
 	 * @unreleased
 	 *
+	 * @param string $accountId
+	 *
 	 * @return bool
 	 */
-	public static function canAddFee() {
+	public static function canAddFee( $accountId ) {
```

Comment:
Instead of threading `$accountId` through `canAddFee`, can we move this logic closer to the `isCountrySupportApplicationFee` check itself? It is the only check that needs it.

Reply:
@Kyle B. Johnson can you suggest something?

Reply:
Since `give_stripe_get_connected_account_options` is pulling off of the session, can we call it directly from the check instead of passing through the `$accountId`?

Reply:
Then we could drop the changes to the `includes` files and the change would be contained to the new code.

Reply:
@kjohnson I will not suggest that because that way this function output depends upon the donation section and we of this function also limit to donation processing context.

Reply:
@ravinderk what if we use the container to resolve a `Modle\AccountDetail` from the repository and inject the `give_stripe_get_connected_account_options()` there:

```php
// Service Provider
$this->bind( AccountDetail::class, function() {
    $accountOptions = give_stripe_get_connected_account_options();
    return ( new Repository\AccountDetail( ) )->getAccouuntDetail( $account_options['stripe_account'] );
});

```

Reply:
@kbjohnson90 I feel that we need a class that returns a result of ApplicationFee in a donation processing context. I added a new singleton class. In this way, we can use the `ApplicationFee` independently.

commit: https://github.com/impress-org/givewp/compare/f3ebb6dcb2ff...ac859d6fa180

```php
class ProcessingDonation {
	/**
	 * @var Models\AccountDetail
	 */
	protected $accountDetail;

	public static function canAddFee() {
		/* @var $donationProcessing self */
		$donationProcessing = give( __CLASS__ );
		$gate               = new ApplicationFee( $donationProcessing->accountDetail );

		return $gate->canAddFee();
	}
	
	public function setAccountDetail() {
		// code
		return $this;
	}
}


// in service provider boot function.
give()->singleton( ProcessingDonation::class,
	function() {
		return ( new ProcessingDonation() )->setAccountDetail();
	}
);
```

Reply:
@ravinderk I like where you are going with this. Would it make sense to inject the account details directly into the ApplicationFee class via the container? I'm not sure if the additional class is necessary - I think we can combine these two as they are really doing one thing.

Reply:
@kbjohnson90 I am loading `AccuntDetail` model into `ApplicationFee` class. it looks like:

```php
class ApplicationFee {
	//code
	
	/**
	 * ApplicationFee constructor.
	 *
	 * @param  Models\AccountDetail  $accountDetail
	 */
	public function __construct( Models\AccountDetail $accountDetail ) {
		$this->accountDetail = $accountDetail;
	}
	
	//code
}
``` 

 `ProcessingDonation` and `ApplicationFee` class seems similar but they are.  `ProcessingDonation` bound to context while the `ApplicationFee` class can be used independently. For some reason `ProcessingDonation::canAddFee` is easily accessible then `ApplicationFee`.
 
 ```php
 // use 1
if ( ProcessingDonation::canAddFee() ) {
    // code
}
 
 // use 2
$accountDetailRepository = give( AccountDetail::class );
$accountId  = give_stripe_get_connected_account_options()['stripe_account'];
$applicationFee = new ApplicationFee( $accountDetailRepository->getAccountDetail( $accountId ) );

if( applicationFee->canAddFee() ){
    // code
}
 
 ```